ts/Stock/Indicators/PC/PCIndicator.ts

```diff
@@ -0,0 +1,239 @@
+                 * [plotOptions.pc.color](#plotOptions.pc.color).
+                 *
+                 * @type {Highcharts.ColorString}
+                 */
+                lineColor: palette.colors[8],
+                /**
+                 * Pixel width of the line.
+                 */
+                lineWidth: 1
+            }
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    } as PCOptions);
+
+    /* *
+    *
+    *  Properties
+    *
+    * */
+
+    public data: Array<PCPoint> = void 0 as any;
+    public options: PCOptions = void 0 as any;
+    public points: Array<PCPoint> = void 0 as any;
+
+    /* *
+    *
+    *  Functions
+    *
+    * */
+
+    public getValues<TLinkedSeries extends LineSeries>(
+        series: TLinkedSeries,
+        params: PCParamsOptions
+    ): (IndicatorValuesObject<TLinkedSeries> | undefined) {
+        var period: number = (params.period as any),
+            xVal: Array<number> = (series.xData as any),
+            yVal: Array<Array<number>> = (series.yData as any),
+            yValLen: number = yVal ? yVal.length : 0,
+            // 0- date, 1-top line, 2-middle line, 3-bottom line
+            PC: Array<Array<number>> = [],
+            // middle line, top line and bottom line
+            ML: number,
+            TL: number,
+            BL: number,
+            date: number,
+            low = 2,
+            high = 1,
+            xData: Array<number> = [],
+            yData: Array<Array<number>> = [],
+            slicedY: Array<Array<number>>,
+            extremes: [number, number],
+            i: number;
+
+        if (yValLen < period) {
+            return;
+        }
+
+        for (i = period; i <= yValLen; i++) {
+            date = xVal[i - 1];
+            slicedY = yVal.slice(i - period, i);
+            extremes = getArrayExtremes(slicedY, low as any, high as any);
+            TL = extremes[1];
+            BL = extremes[0];
+            ML = (TL + BL) / 2;
+            PC.push([date, TL, ML, BL]);
+            xData.push(date);
+            yData.push([TL, ML, BL]);
+        }
+
+        return {
+            values: PC,
+            xData: xData,
+            yData: yData
+        } as IndicatorValuesObject<TLinkedSeries>;
+    }
+}
+
+/* *
+*
+*   Prototype Properties
+*
+* */
+
+interface PCIndicator {
+    getTranslatedLinesNames: Highcharts.MultipleLinesMixin[
+        'getTranslatedLinesNames'
+    ];
+    linesApiNames: Highcharts.MultipleLinesMixin['linesApiNames'];
+    nameBase: string;
+    nameComponents: Array<string>;
+    pointArrayMap: Array<string>;
+    pointClass: typeof PCPoint;
+    pointValKey: string;
+}
+
+extend(PCIndicator.prototype, merge(MultipleLinesMixin, {
+    pointArrayMap: ['top', 'middle', 'bottom'],
+    pointValKey: 'middle',
+    nameBase: 'Price Channel',
+    nameComponents: ['period'],
+    linesApiNames: ['topLine', 'bottomLine']
+}));
```

Comment:
The mixin has to be manually assigned, to make it clear for future maintenance, what actually gets replaced in the class pattern. So it should become:
```ts
extend(PCIndicator.prototype, {
    getTranslatedLinesNames: MultipleLinesMixin.getTranslatedLinesNames,
    …
    translate: MultipleLinesMixin.translate
});
```
As you will notice, some values/functions like `linesApiNames` are actually not taken from the mixin like the interface suggests, but are redefined in the class pattern or the extend call.

Reply:
So, could you please suggest how to deal with these linesApiNames? 
Since leaving such a configuration causes errors:
```ts
extend(PCIndicator.prototype, {
    getTranslatedLinesNames: MultipleLinesMixin.getTranslatedLinesNames,
    pointArrayMap: ['top', 'middle', 'bottom'],
    pointValKey: 'middle',
    nameBase: 'Price Channel',
    nameComponents: ['period'],
    linesApiNames: ['topLine', 'bottomLine'],
    translate: MultipleLinesMixin.translate
});
```

Reply:
You have to assign all the parts of the mixin, that are not defined in the PCIndicator class pattern or extend call. As far as I can see, it is missing: `drawGraph`, `toYData`.